modules/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageRetryingInputStream.java

```diff
@@ -126,13 +154,23 @@ private InputStream openStream() throws IOException {
             } catch (RetryHelper.RetryHelperException e) {
                 throw StorageException.translateAndThrow(e);
             }
-        } catch (StorageException e) {
-            if (e.getCode() == 404) {
+        } catch (StorageException storageException) {
+            if (storageException.getCode() == RestStatus.NOT_FOUND.getStatus()) {
                 throw addSuppressedExceptions(
-                    new NoSuchFileException("Blob object [" + blobId.getName() + "] not found: " + e.getMessage())
+                    new NoSuchFileException("Blob object [" + blobId.getName() + "] not found: " + storageException.getMessage())
                 );
             }
-            throw addSuppressedExceptions(e);
+            if (storageException.getCode() == RestStatus.REQUESTED_RANGE_NOT_SATISFIED.getStatus()) {
+                throw addSuppressedExceptions(
+                    new RequestedRangeNotSatisfiedException(
+                        blobId.getName(),
+                        start,
```

Comment:
Should here (and below line) be `Math.addExact(start, currentOffset)` instead of just `start`?

Reply:
thank you @kingherc